src/to_substrait.cpp

```diff
@@ -879,37 +879,53 @@ substrait::RelCommon *DuckDBToSubstrait::CreateOutputMapping(vector<int32_t> vec
 	return rel_common;
 }
 
+bool DuckDBToSubstrait::IsPassthroughProjection(LogicalProjection &dproj, idx_t child_column_count) {
+	// check if the projection is just pass through of input columns with no reordering
+	if (child_column_count > dproj.expressions.size()) {
+		return false;
+	}
+	idx_t exp_col_idx = 0;
+	for (auto &dexpr : dproj.expressions) {
+		if (dexpr->type != ExpressionType::BOUND_REF) {
+			return false;
+		}
+		auto &dref = dexpr.get()->Cast<BoundReferenceExpression>();
+		if (dref.index != exp_col_idx) {
+			return false;
+		}
+		exp_col_idx++;
+	}
+	return child_column_count == exp_col_idx;
+}
+
+bool DuckDBToSubstrait::NeedsOutputMapping(LogicalProjection &dproj, idx_t child_column_count) {
```

Comment:
Is this the same as IsPassthroughProjection?

Reply:
it is slightly different.

Reply:
It isn't, there are two lines that are different.